aten/src/ATen/native/cuda/MemoryAccess.cuh

```diff
@@ -387,4 +387,51 @@ inline int can_vectorize_up_to(array_t pointers) {
   return result;
 }
 
+// This is only used in host, but we will wrap this into some templates
+// which is C10_HOST_DEVICE, so we have to make this C10_HOST_DEVICE
+// in order to compile
+template<typename scalar_t>
+inline C10_HOST_DEVICE int can_vectorize_elementwise_up_to(const char *pointer) {
```

Comment:
it seems like instead of copy-pasting `can_vectorize_up_to` you can add a template argument to it to indicate max value you can return (4 or 8)